### PR TITLE
chore: remove @openfeature/server-sdk as a runtime dependency

### PR DESCRIPTION
**Pull request by greghuels**
## What

At a high-level, this PR removes `@openfeature/server-sdk` as a runtime dependency.

- **Custom EventEmitter** — replaces `OpenFeatureEventEmitter` from `@openfeature/server-sdk` with a custom `NodeProviderEventEmitter` built on Node's native `EventEmitter`. Zero `@openfeature` imports in compiled `.js`.
- **Bundled type declarations** — `dts-bundle-generator` inlines all `@openfeature/core` and `@openfeature/server-sdk` types into a single self-contained `index.d.ts` with zero `@openfeature` references. Necessary for SSI where `dd-trace`'s `node_modules` can't reach the customer's `@openfeature/server-sdk`.

## Why

`dd-trace-js` plans to vendor the OpenFeature provider to work around `@datadog/openfeature-node-server` having `@openfeature/server-sdk` as a peer dependency, which fails under SSI ([dd-trace-js#9570](https://github.com/DataDog/dd-trace-js/pull/9570)). This PR removes the `@openfeature/server-sdk` runtime dependency so `dd-trace-js` can instead declare `@datadog/openfeature-node-server` as a regular dependency.

This approach has several benefits:
- Reduces build complexity in dd-trace-js
- Allows for patch fixes to automatically reach customers, assuming dd-trace-js specifies this library as a patch-level range in its package.json.

### Why `build:types` is needed

`tsc` alone produces `.d.ts` files that preserve external imports:

```typescript
// cjs/provider.d.ts (tsc output)
import type { Provider, ResolutionDetails, ... } from "@openfeature/server-sdk";
import type { ServerProviderEvents, ... } from "@openfeature/core";
```

Under SSI, `dd-trace` is installed outside the application's `node_modules`, so these `.d.ts` files live in `dd-trace`'s `node_modules` tree. TypeScript resolves the `@openfeature/server-sdk` import by walking up from the `.d.ts` file's location — it never reaches the customer's `node_modules` where their `@openfeature/server-sdk` is actually installed:

```
/opt/datadog/dd-trace-js/                          ← dd-trace lives here
  node_modules/
    @datadog/openfeature-node-server/
      cjs/provider.d.ts                             ← .d.ts is here, can't resolve @openfeature/server-sdk

/home/user/my-app/                                  ← customer's app
  node_modules/
    @openfeature/server-sdk/                        ← customer's SDK is here, unreachable
```

`build:types` uses `dts-bundle-generator` to inline all `@openfeature/*` types into a single self-contained `index.d.ts` with zero external `@openfeature` imports, so the types resolve regardless of where `dd-trace` is installed.

See https://github.com/DataDog/dd-trace-js/pull/9570 for the SSI context.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,12 @@
     "webpack": "5.106.2"
   },
   "packageManager": "yarn@4.10.3",
+  "resolutions": {
+    "tar@npm:7.5.11": "npm:^7.5.22",
+    "minimatch@npm:10.1.1": "npm:^10.2.6",
+    "js-yaml@npm:4.1.1": "npm:^4.3.1",
```
Comment (greghuels) [inline package.json:52]: Pre-existing security fixes that would have been overwritten in the yarn.lock. This codifies the install so that `yarn.lock` regeneration doesn't affect manually upgraded dependencies.

File: package.json [hunk 2]
```diff
@@ -46,6 +46,12 @@
...
+    "typescript@npm:>=5.0.2": "npm:5.9.3"
```
Comment (greghuels) [inline package.json:53]: dts-bundle-generator's typescript >=5.0.2 dependency caused yarn to resolve TS 7.0.2

File: packages/node-server/package.json [hunk 3]
```diff
@@ -25,27 +25,27 @@
...
   "scripts": {
-    "prepack": "yarn build",
+    "prepack": "yarn build && node scripts/verify-no-openfeature-dep.js",
     "pack": "yarn pack",
-    "build": "run-p build:cjs build:esm",
+    "build": "run-p build:cjs build:esm && yarn build:types",
     "build:cjs": "rm -rf cjs && tsc -p tsconfig.cjs.json && yarn replace-build-env cjs",
     "build:esm": "rm -rf esm && tsc -p tsconfig.esm.json && yarn replace-build-env esm",
+    "//build:types": "Bundles @openfeature/* types into a self-contained index.d.ts with no external imports, required for SSI where dd-trace's node_modules can't reach the customer's @openfeature/server-sdk. See https://github.com/DataDog/dd-trace-js/pull/9570",
+    "build:types": "trap 'rm -f src/.dts-entry.ts' EXIT; printf '/// <reference path=\"./global.d.ts\" />\\nexport * from \"./index\"\\n' > src/.dts-entry.ts && dts-bundle-generator -o index.d.ts src/.dts-entry.ts --external-inlines @openfeature/core @openfeature/server-sdk --no-banner --export-referenced-types false",
```
Comment (greghuels) [inline packages/node-server/package.json:34]: `tsc` produces `.d.ts` files that preserve external imports (`import type { ... } from "@openfeature/server-sdk"`). Under SSI, `dd-trace` is installed outside the application's `node_modules`, so TypeScript can't resolve those imports — the `.d.ts` files live in `dd-trace`'s `node_modules` tree and can't reach the customer's `@openfeature/server-sdk`.

`dts-bundle-generator` inlines all `@openfeature/*` types into a single self-contained `index.d.ts` with zero external imports, so the types resolve regardless of where `dd-trace` is installed.

See https://github.com/DataDog/dd-trace-js/pull/9570 for the SSI context.

File: packages/node-server/src/provider-event-emitter.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  EventHandler,
+  Logger,
+  ProviderEventEmitter,
+  ServerProviderEvents,
+} from '@openfeature/core'
+
+/**
+ * OpenFeature-compatible event emitter that does not require an OpenFeature SDK at runtime.
+ */
+export class NodeProviderEventEmitter implements ProviderEventEmitter<ServerProviderEvents> {
```
Comment (leoromanovsky) [inline packages/node-server/src/provider-event-emitter.ts:14]: Please add tests for add/remove/remove-all, duplicate handlers, async rejection, etc.

File: packages/node-server/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,598 @@
+import { Channel } from 'node:diagnostics_channel';
```
Comment (leoromanovsky) [inline packages/node-server/index.d.ts:1]: Please add a test to exercise type-checks and runs against minimum and current supported OpenFeature SDK versions.

**Review by leoromanovsky — COMMENTED** [review]
lgtm! Please increase test coverage in the support version range.

File: packages/node-server/src/provider-event-emitter.ts (hunk 4 above)
Comment (greghuels) [reply packages/node-server/src/provider-event-emitter.ts:14]: Yeah, good call. This is implicitly tested with integration tests, but the class itself could use some unit tests.

File: packages/node-server/index.d.ts (hunk 5 above)
Comment (greghuels) [reply packages/node-server/index.d.ts:1]: I'll update in https://github.com/DataDog/openfeature-js-client/pull/357, since I already have typecheck tests there as well.

Comment (leoromanovsky) [reply packages/node-server/index.d.ts:1]: Ok I see now, that's fine.

**Review by leoromanovsky — APPROVED** [review]
Oops meant to pre-approved pending the two comments.

File: packages/node-server/src/provider-event-emitter.ts (hunk 4 above)
Comment (greghuels) [reply packages/node-server/src/provider-event-emitter.ts:14]: updated!

File: packages/node-server/index.d.ts (hunk 5 above)
Comment (greghuels) [reply packages/node-server/index.d.ts:1]: Merged into this PR and now running against the min supported openfeature version
